### PR TITLE
manifest: Updated sdk-connectedhomeip to the new revision

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Regular uppmerge of sdk-connectedhomeip including necessary
changes in sdk-nrf:
* Changed API in light-switch binding handler
* Updated documentation
* Regenerated ZAP files

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: applications/matter_weather_station/src/weather-station.matter [hunk 1]
```diff
@@ -157,7 +157,12 @@ server cluster Basic = 40 {
   readonly attribute int32u softwareVersion = 9;
   readonly attribute char_string<64> softwareVersionString = 10;
   readonly attribute char_string<16> manufacturingDate = 11;
+  readonly attribute char_string<32> partNumber = 12;
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/src/weather-station.matter:160]: Are those changes needed? Last time I configured these attributes to match what we test in SVE.

File: applications/matter_weather_station/src/weather-station.matter [hunk 2]
```diff
@@ -575,12 +586,45 @@ server cluster NetworkCommissioning = 49 {
   }
 
   command access(invoke: administer) ScanNetworks(ScanNetworksRequest): ScanNetworksResponse = 0;
+  command access(invoke: administer) AddOrUpdateWiFiNetwork(AddOrUpdateWiFiNetworkRequest): NetworkConfigResponse = 2;
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/src/weather-station.matter:589]: Will we support WiFi in the weather station?

File: doc/nrf/links.txt [hunk 3]
```diff
@@ -94,8 +94,8 @@
 .. _`SMP over Bluetooth`: https://github.com/apache/mynewt-mcumgr/blob/master/transport/smp-bluetooth.md
 
 .. _`Matter GitHub repository`: https://github.com/project-chip/connectedhomeip
-.. _`Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/tree/bb3c7570/src/android/CHIPTool
-.. _`other controller setups`: https://github.com/nrfconnect/sdk-connectedhomeip/tree/bb3c7570/src/controller
+.. _`Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/tree/708685f4/src/android/CHIPTool
```
Comment (Damian-Nordic) [inline doc/nrf/links.txt:97]: It's supposed to be hash in sdk-connectedhomeip, not upstream.

Comment (kkasperczyk-no) [reply doc/nrf/links.txt:97]: Right, will update it after merging new SHA

File: applications/matter_weather_station/src/weather-station.matter (hunk 1 above)
Comment (kkasperczyk-no) [reply applications/matter_weather_station/src/weather-station.matter]: Ok, I automatically aligned it to other samples, but we can also keep it consistent with SVE. Did you mean only basic cluster or all? I reverted changes for Basic cluster.

Comment (Damian-Nordic) [reply applications/matter_weather_station/src/weather-station.matter]: I mean all. We don't implement diagnostic logs, Thread metrics etc. If they are enabled in other samples, we should rather disable them there.

Comment (kkasperczyk-no) [reply applications/matter_weather_station/src/weather-station.matter]: I think we should work on creating some kind of array of supported features for our samples then. Reviewing all NCS vs upstream differences in clusters and deciding should we support it or not on every upmerge is not that easy. I can try to revert some configuration, but honestly I'm not sure which part of it is useless and which might be used somewhere under the hood.

**Review by annwoj — APPROVED** [review]

File: applications/matter_weather_station/src/weather-station.matter [hunk 4]
```diff
@@ -1153,7 +1158,7 @@ endpoint 0 {
     callback attribute acceptedCommandList;
     callback attribute attributeList;
     ram      attribute featureMap;
-    ram      attribute clusterRevision default = 1;
+    ram      attribute clusterRevision default = 3;
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/src/weather-station.matter:1161]: I don't think it's right. The spec defines just a single revision for this cluster.

File: applications/matter_weather_station/src/weather-station.matter [hunk 5]
```diff
@@ -1202,19 +1207,19 @@ endpoint 0 {
...
-    callback attribute maxNetworks;
+    ram      attribute maxNetworks;
     callback attribute networks;
-    callback attribute scanMaxTimeSeconds;
-    callback attribute connectMaxTimeSeconds;
-    callback attribute interfaceEnabled;
-    callback attribute lastNetworkingStatus;
-    callback attribute lastNetworkID;
-    callback attribute lastConnectErrorValue;
+    ram      attribute scanMaxTimeSeconds;
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/src/weather-station.matter:1212]: All those are handled by callbacks, so don't require ram storage.

File: samples/matter/light_switch/README.rst [hunk 6]
```diff
@@ -343,8 +343,8 @@ After building this and the :ref:`Matter Light Bulb <matter_light_bulb_sample>`
    See :ref:`matter_light_switch_sample_remote_control_commissioning` for more information.
    During the commissioning process, write down the values for the light switch node ID and the light bulb node ID (or IDs, if you are using more than one light bulb).
    These IDs are going to be used in the next steps (*<light_switch_node_ID>* and *<light_bulb_node_ID>*, respectively).
-#. Use the :doc:`CHIP Tool <matter:chip_tool_guide>` ("Writing ACL to the ``accesscontrol`` cluster" section) to add proper ACL for the light bulb device.
-   Depending on the number of the light bulb devices you are using, use one of the following commands, with *<light_switch_node_ID>* and *<light_bulb_node_ID>* values from the previous step about commissioning:
+#. If you are using only one light bulb device, use the :doc:`CHIP Tool <matter:chip_tool_guide>` ("Writing ACL to the ``accesscontrol`` cluster" section) to add proper ACL for the light bulb device.
```
Comment (Damian-Nordic) [inline samples/matter/light_switch/README.rst:346]: Is this intentional change? Looks like we have duplicated condition "If you are using only one light bulb..."

**Review by Damian-Nordic — APPROVED** [review]
A few minors found, but LGTM overall

**Comment by kkasperczyk-no** [comment]
@gmarull why did you merge it? Not all comments were resolved and also on-desk testing was not yet done.

**Comment by gmarull** [comment]
> @gmarull why did you merge it? Not all comments were resolved and also on-desk testing was not yet done.

Then please request changes instead. Please send a follow-up PR.

**Comment by kkasperczyk-no** [comment]
So far the process was to ask for merging PRs on the dedicated channel, so I assumed PR will not be merged without my direct request. I was not aware that it changed for the release period, but ok I will block my PR with DNM next time then.

**Pull request merged**